github/commands/pull_request.py

```diff
@@ -101,17 +101,20 @@ def fetch_and_checkout_pr(self, branch_name=None):
             self.pr["head"]["ref"]
         )
 
-        if branch_name:
-            self.window.status_message("Creating local branch for PR...")
-            self.git(
-                "branch",
-                branch_name,
-                self.pr["head"]["sha"]
-            )
+        if not branch_name:
+            branch_name = self.pr["head"]["sha"]
+
+        self.window.status_message("Creating local branch for PR...")
+        self.git(
+            "branch",
+            branch_name,
+            self.pr["head"]["sha"]
+        )
 
         self.window.status_message("Checking out PR...")
-        self.checkout_ref(branch_name or self.pr["head"]["sha"])
+        self.checkout_ref(branch_name)
         util.view.refresh_gitsavvy_interfaces(self.window, refresh_sidebar=True)
+        sublime.set_timeout(lambda: self.set_upstream_for_branch(branch_name))
```

Comment:
Avoid `set_timeout` here. Why is that needed. Also call `refresh_gitsavvy` at the end.